### PR TITLE
Implement 'channel' argument for git-extraction via process addons.

### LINKED ISSUE #10651: Implement a management script that extracts versions into git-storage manually
We're going to need a script that processes versions that haven't been extracted into git yet.

It should feature the following modes:

* Process all listed versions (only touching versions that haven't been extracted yet)
* Process all unlisted versions (only touching versions that haven't been extracted yet)
* Process one version explicitly (e.g when problems happen and we need manual extraction)
* Force extraction for listed or unlisted versions (e.g when a new storage format gets introduced - like in #10043)

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This will allow selectively (re-)extract only listed or unlisted addons.

Fixes #10651

### CODE REVIEW

File: src/olympia/addons/management/commands/process_addons.py [hunk 1]
```diff
@@ -213,6 +214,13 @@ def add_arguments(self, parser):
             default=100,
             help='Split the add-ons into X size chunks. Default 100.')
 
+        parser.add_argument(
+            '--channel',
+            action='store',
+            dest='channel',
+            type=str,
+            choices=('listed', 'unlisted'))
```
Comment (eviljeff) [inline src/olympia/addons/management/commands/process_addons.py]: can you add help text?

File: src/olympia/addons/management/commands/process_addons.py [hunk 2]
```diff
@@ -125,13 +125,14 @@ def get_recalc_needed_filters():
     'extract_webextensions_to_git_storage': {
         'method': migrate_webextensions_to_git_storage,
         'qs': [
-            Q(_current_version__files__is_webextension=True,
-              type__in=(
+            Q(type__in=(
                   # Ignoring legacy add-ons and lightweight themes
                   amo.ADDON_EXTENSION, amo.ADDON_STATICTHEME,
                   amo.ADDON_DICT, amo.ADDON_LPAPP)) |
             Q(type=amo.ADDON_SEARCH)
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/addons/management/commands/process_addons.py]: Ah yeah, we don't really have any of the others lying around on prod anymore but I think that change can be done in a separate cleanup step. I'd rather play it safe when talking about running extraction for a whole lot more addons than we had in the past.

[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/addons/management/commands/process_addons.py]: Fixed in 99f03bd98d

File: src/olympia/addons/management/commands/process_addons.py (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/addons/management/commands/process_addons.py]: Added in 99f03bd98d

**Pull request merged**